atom/src/defaultvaluebehavior.cpp

```diff
@@ -212,8 +211,6 @@ handlers[] = {
 PyObject*
 Member::default_value( CAtom* atom )
 {
-    if( get_default_value_mode() >= sizeof( handlers ) )
-        return no_op_handler( this, atom );  // LCOV_EXCL_LINE
     return handlers[ get_default_value_mode() ]( this, atom );
```

Comment:
Here you basically eliminated bound checks. It may be better to use std::array::at to benefit from the stdlib bound checks.

Reply:
The check is done when it parses the enum value, but I agree it's still possible to go OOB if the mode bits somehow get misparsed or that check is bypassed.

My goal of originally looking at this was to try and delete if's... I'd rather go back to re-adding no_op_handlers than add another if.

Reply:
`std::array<handler, 8>` doesn't give a compile error if you give less than 8 entries so I'd rather just go back to https://github.com/nucleic/atom/pull/235/commits/8b32c1b90ee0f47b41662cf7e679a2ce32cbb271 with maybe an additional static_assert check on the sentinel?

Reply:
Can we statically assert that we do not have more enum variants than we have handlers?  if we can do that we should be good.